iothub/service/src/Http/HttpMessageHelper.cs

```diff
@@ -91,36 +92,47 @@ internal static async Task<T> DeserializeResponseAsync<T>(HttpResponseMessage re
         /// <param name="eTag">The If-Match header value to sanitize before adding.</param>
         /// <exception cref="ArgumentNullException">Thrown when the provided <paramref name="requestMessage"/> or <paramref name="requestMessage"/> is null.</exception>
         /// <exception cref="ArgumentException">Thrown if the <paramref name="eTag"/> is empty or white space.</exception>
-        /// <exception cref="IotHubException">
-        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
-        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
-        /// error cases, see <see cref="Common.Exceptions"/>.
-        /// </exception>
-        /// <exception cref="HttpRequestException">
-        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
-        /// certificate validation.
-        /// </exception>
-        public static void InsertETag(HttpRequestMessage requestMessage, string eTag)
+        public static void ConditionallyInsertETag(HttpRequestMessage requestMessage, string eTag)
```

Comment:
Why have this overload rather than force all callers to be explicit?

Reply:
To not break existing clients

Reply:
Until you finish migration, then it can go away?

Reply:
Correct